lib/dev.js

```diff
@@ -71,7 +72,9 @@ module.exports = async function dev (sourceDir, cliOptions = {}) {
   }
 
   const compiler = webpack(config)
-  const port = cliOptions.port || options.siteConfig.port || 8080
+  portfinder.basePort = cliOptions.port || options.siteConfig.port || 8080
+  const port = await portfinder.getPortPromise()
+    .catch(err => console.log(err))
```

Comment:
no need to catch here, this should propagate up the call stack and be logged anyway.